app/scripts/models/reliers/oauth.js

```diff
@@ -227,12 +230,21 @@ define(function (require, exports, module) {
      * @returns {Boolean}
      */
     wantsKeys () {
-      return !! (this._config && this._config.scopedKeysEnabled && this._validateKeyScopeRequest());
+      return !! (this._config && this._config.scopedKeysEnabled && this.has('keysJwk'));
     },
 
+
     /**
-     * Validate the requested scope with the relier redirect uri.
-     * At least one valid match must be found to successfully validate
+     * Perform additional validation for scopes that have encryption keys.
+     *
+     * If the relier is requesting keys, we check their redirect URI against
+     * against an explicit allowlist and throw an error if there's anything
+     * even slightly unexpected about the request.
+     *
+     * This provides an extra line of defence against us sending the keys
+     * somewhere unintended as a result of e.g. a config error or a
+     * badly-behaved server.
+     *
```

Comment:
I did a lot of talking to myself in comments while working with this refactor, figured I may as well leave them in place even though the code didn't end up moving around as much as I thought it might.